src/object_.d

```diff
@@ -2528,16 +2819,18 @@ version(unittest) unittest
 
 void destroy(T)(ref T obj) if (is(T == struct))
 {
-    typeid(T).destroy( &obj );
-    auto buf = (cast(ubyte*) &obj)[0 .. T.sizeof];
-    auto init = cast(ubyte[])typeid(T).init();
-    if(init.ptr is null) // null ptr means initialize to 0s
-        buf[] = 0;
-    else
-        buf[] = init[];
+    _destructRecurse(obj);
+    () @trusted {
+        auto buf = (cast(ubyte*) &obj)[0 .. T.sizeof];
+        auto init = cast(ubyte[])typeid(T).init();
+        if (init.ptr is null) // null ptr means initialize to 0s
+            buf[] = 0;
```

Comment:
Can you please follow up and replace this with memset for `T.sizeof >= 4`.
[Issue 14458 - very slow ubyte[] assignment (dmd doesn't use memset)](http://issues.dlang.org/show_bug.cgi?id=14458